### ISSUE TITLE
Crashes on launch in M1 macOS 12.4

### ISSUE THREAD

**Issue by apetresc** [schadfield/shogi-explorer#1]
I've just downloaded the latest 4.1.0 release on my M1 Mac Mini running macOS 12.4. After extracting and launching the .app, it immediately crashes before displaying any windows with:

```
The application cannot be opened for an unexpected reason, error=Error Domain=RBSRequestErrorDomain Code=5 "Launch failed." UserInfo={NSLocalizedFailureReason=Launch failed., NSUnderlyingError=0x6000006596b0 {Error Domain=NSPOSIXErrorDomain Code=111 "Unknown error: 111" UserInfo={NSLocalizedDescription=Launchd job spawn failed}}}
```

Let me know if there's any other diagnostics I can provide :) I'm running OpenJDK 18 if that matters.

**Comment by apetresc** [comment]
Okay, if I open up the `.app` package and run the `Shogi Explorer.app/Contents/Resources/Java/Shogi Explorer-4.1.0-runnable.jar` directly, it works fine. So it's something wrong with the .app packaging that's preventing it from running.

**Comment by schadfield** [comment]
Could you try it with OpenJDK 17? I have never tested it with anything newer.

**Comment by apetresc** [comment]
The .jar file works with both JDK 17 and JDK 18 (I just tested with both) if I just run `java -jar /Applications/Shogi\ Explorer.app/Contents/Resources/Java/Shogi\ Explorer-4.1.0-runnable.jar`. The problem isn't the Java code per-se, it's the .app packaging somehow.

I suspect launchd isn't even getting to the point of _attempting_ to run Java because when I ran it directly, it still gave me the prompt about accepting the unsigned executable, which only happens on first launch and hadn't happened when I ran the .app. So the crash is caused by something in the Info.plist most likely.

**Comment by schadfield** [comment]
Unfortunately I have sold my MacBook so I currently have no way of testing on macOS. I am going to have to rely on users such as yourself to help me out!

**Comment by schadfield** [comment]
This might be a similar issue as I am now building the packages on Windows. There are some permissions which need changing on a couple of files.

[https://forum.unity.com/threads/issue-opening-unity-mac-applications-on-mac-big-sur-os-from-unidentified-developer.1026697/#post-6650740](https://forum.unity.com/threads/issue-opening-unity-mac-applications-on-mac-big-sur-os-from-unidentified-developer.1026697/#post-6650740)

**Comment by schadfield** [comment]
Maybe:

chmod +x Shogi\ Explorer.app/Contents/MacOS/universalJavaApplicationStub
xattr -r -d com.apple.quarantine Shogi\ Explorer.app/

**Comment by apetresc** [comment]
Yes! That did solve it, the .app launches successfully now. (I think it was the permission on the stub, because if I understand correctly, the `com.apple.quarantine` just removes the need for accepting the unidentified developer signing cert, which you are prompted to do manually through the Security prefPane anyway).

**Comment by schadfield** [comment]
Cool. I guess if I build the packages in a Linux virtual machine the executable bit will be set correctly. I will do that going forward.

**Comment by schadfield** [comment]
Released 4.1.1 where all the packages are built on Debian stable.
